metadata-ingestion/src/datahub/ingestion/source/glue.py

```diff
@@ -0,0 +1,211 @@
+import logging
+import time
+from dataclasses import dataclass
+from typing import Iterable, Optional
+
+import boto3
+from datahub.configuration import ConfigModel
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.source.metadata_common import MetadataWorkUnit
+from datahub.metadata.com.linkedin.pegasus2avro.common import AuditStamp, Status
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import DatasetSnapshot
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    ArrayTypeClass,
+    BooleanTypeClass,
+    BytesTypeClass,
+    DateTypeClass,
+    NullTypeClass,
+    NumberTypeClass,
+    SchemaField,
+    SchemaFieldDataType,
+    StringTypeClass,
+    TimeTypeClass,
+    UnionTypeClass,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    SchemaMetadata,
+)
+from datahub.metadata.schema_classes import DatasetPropertiesClass, OwnershipClass, OwnerClass, OwnershipTypeClass, \
+    AuditStampClass, MapTypeClass
+
+logger = logging.getLogger(__name__)
+
+
+def get_column_type(field_type: str) -> SchemaFieldDataType:
+    _field_type_mapping = {
+        "array": ArrayTypeClass,
+        "bigint": NumberTypeClass,
+        "binary": BytesTypeClass,
+        "boolean": BooleanTypeClass,
+        "char": StringTypeClass,
+        "date": DateTypeClass,
+        "decimal": NumberTypeClass,
+        "double": NumberTypeClass,
+        "float": NumberTypeClass,
+        "int": NumberTypeClass,
+        "integer": NumberTypeClass,
+        "interval": TimeTypeClass,
+        "long": NumberTypeClass,
+        "map": MapTypeClass,
+        "null": NullTypeClass,
+        "set": ArrayTypeClass,
+        "smallint": NumberTypeClass,
+        "string": StringTypeClass,
+        "struct": MapTypeClass,
+        "timestamp": TimeTypeClass,
+        "tinyint": NumberTypeClass,
+        "union": UnionTypeClass,
+        "varchar": StringTypeClass,
+    }
+
+    if field_type in _field_type_mapping.keys():
+        type_class = _field_type_mapping[field_type]
+    elif field_type.startswith("array"):
+        type_class = ArrayTypeClass
+    elif field_type.startswith("map") or field_type.startswith("struct"):
+        type_class = MapTypeClass
+    elif field_type.startswith("set"):
+        type_class = ArrayTypeClass
+    # TODO better handling of unknown type
+    else:
+        type_class = NullTypeClass
+    data_type = SchemaFieldDataType(type=type_class())
+    return data_type
+
+
+# TODO: set this @dataclass
+class GlueSourceConfig(ConfigModel):
+    env: str = "PROD"
+    aws_access_key_id: Optional[str] = None
```

Comment:
Would be nice to have the schema_pattern and table_pattern configurable. Otherwise you're forced to ingest all the Glue tables/ databases, which might not be reasonable.